mesonbuild/dependencies/base.py

```diff
@@ -956,16 +956,25 @@ def __init__(self, name, environment, kwargs, language=None):
             return
 
         modules = kwargs.get('modules', [])
+        cm_path = kwargs.get('module_path', [])
+        cm_args = kwargs.get('cmake_args', [])
         if not isinstance(modules, list):
             modules = [modules]
-        self._detect_dep(name, modules)
+        if not isinstance(cm_path, list):
+            cm_path = [cm_path]
+        if not isinstance(cm_args, list):
+            cm_args = [cm_args]
+        cm_path = [x if os.path.isabs(x) else os.path.join(environment.get_source_dir(), x) for x in cm_path]
+        if len(cm_path) > 0:
+            cm_args += ['-DCMAKE_MODULE_PATH={}'.format(';'.join(cm_path))]
+        self._detect_dep(name, modules, cm_args)
 
     def __repr__(self):
         s = '<{0} {1}: {2} {3}>'
         return s.format(self.__class__.__name__, self.name, self.is_found,
                         self.version_reqs)
 
-    def _detect_dep(self, name, modules):
+    def _detect_dep(self, name: str, modules: list, args: list):
```

Comment:
these shouldn't be `list`, they should be `typing.List[str]`, right?

Reply:
yes assuming there are only strings inside:
```suggestion
    def _detect_dep(self, name: str, modules: List[str], args: List[str]):
```

Reply:
Also fixed with the last commit.

Reply:
The usual convention for type hinting is as follows, to avoid excess verbosity in function declarations. However, whichever way anyone prefers is fine of course:

```python
from typing import List
...
 def _detect_dep(self, name: str, modules: List[str], args: List[str]):
...
```

Reply:
Sure. Updated in the last commit.